src/kv-queue/decorator.ts

```diff
@@ -0,0 +1,13 @@
+import { MetadataHelper } from '../metadata/mod.ts';
+import { queueListenerMetadataKey } from './constants.ts';
+
+export const OnQueueMessage = (channel: string): MethodDecorator => {
```

Comment:
WDYT of having an `InjectQueue` to inject a publisher?

- ref: https://docs.nestjs.com/techniques/queues#producers

Reply:
It made sense for Nest as they are using Bull and it allows multiple connection and/or queue separation.

It will probably make sense when we implement other queue system, but we may have specialised module for them ?

Still using the same "OnQueueMessage" to subscribe but each queue system will can have its own provider to send messages

Reply:
Yeah, it'll make more sense when we implement the multi-queue feature.